From b692a5edb40b6ddb277efd5cd812bf5009972bbe Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 30 Sep 2021 18:32:29 +0000
Subject: [PATCH] GCC: fix name clash of blink::RegionCaptureCropId

blink::RegionCaptureCropId is a typedef and at the same time
RegionCaptureCropId() is a method in class blink::Element.
GCC errors out, because it cannot distinguish between both. Rename
the method to GetRegionCaptureCropId() to avoid the clash.
---
 third_party/blink/renderer/core/dom/element.cc | 6 +++---
 third_party/blink/renderer/core/dom/element.h  | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/third_party/blink/renderer/core/dom/element.cc b/third_party/blink/renderer/core/dom/element.cc
index a4ffa3d..9cb3952 100644
--- a/third_party/blink/renderer/core/dom/element.cc
+++ b/third_party/blink/renderer/core/dom/element.cc
@@ -3625,7 +3625,7 @@ void Element::SetNeedsCompositingUpdate() {
 }
 
 RegionCaptureCropId Element::MarkWithRegionCaptureCropId() {
-  if (RegionCaptureCropId().is_empty()) {
+  if (GetRegionCaptureCropId().is_empty()) {
     EnsureElementRareData().SetRegionCaptureCropId(
         base::UnguessableToken::Create());
 
@@ -3636,10 +3636,10 @@ RegionCaptureCropId Element::MarkWithRegionCaptureCropId() {
       GetLayoutObject()->SetShouldDoFullPaintInvalidation();
     }
   }
-  return RegionCaptureCropId();
+  return GetRegionCaptureCropId();
 }
 
-RegionCaptureCropId Element::RegionCaptureCropId() const {
+RegionCaptureCropId Element::GetRegionCaptureCropId() const {
   return HasRareData() ? GetElementRareData()->RegionCaptureCropId()
                        : base::UnguessableToken::Null();
 }
diff --git a/third_party/blink/renderer/core/dom/element.h b/third_party/blink/renderer/core/dom/element.h
index 334cfcd..1263fe7 100644
--- a/third_party/blink/renderer/core/dom/element.h
+++ b/third_party/blink/renderer/core/dom/element.h
@@ -564,7 +564,7 @@ class CORE_EXPORT Element : public ContainerNode, public Animatable {
   RegionCaptureCropId MarkWithRegionCaptureCropId();
 
   // Returns a null token if not marked for capture.
-  RegionCaptureCropId RegionCaptureCropId() const;
+  RegionCaptureCropId GetRegionCaptureCropId() const;
 
   ShadowRoot* attachShadow(const ShadowRootInit*, ExceptionState&);
 
-- 
2.32.0

